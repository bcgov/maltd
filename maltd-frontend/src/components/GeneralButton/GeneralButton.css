.general-button {
  width: 200px;
}
