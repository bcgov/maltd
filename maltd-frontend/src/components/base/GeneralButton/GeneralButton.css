.general-button {
  width: 150px !important;
  height: 40px !important;
  color: #ffffff !important;
}

/* Bootstrap .btn CSS override */
.btn:focus {
  box-shadow: none !important;
}

/* Bootstrap .btn-primary CSS override */
.btn-primary {
  background-color: #4788c4 !important;
  border-color: #979797 !important;
  opacity: 0.75 !important;
}

.btn-primary:active,
.btn-primary:hover,
.btn-primary:focus {
  background-color: #4788c4 !important;
  border-color: #979797 !important;
  opacity: 1 !important;
}

.btn-primary:disabled,
.btn-primary.disabled,
.btn-primary:visited {
  background-color: #4788c4 !important;
  border-color: #979797 !important;
  opacity: 0.5 !important;
}

/* Bootstrap .btn-danger CSS override */
.btn-danger {
  background-color: #dc3545 !important;
  border-color: #979797 !important;
  opacity: 0.75 !important;
}

.btn-danger:active,
.btn-danger:hover,
.btn-danger:focus {
  background-color: #dc3545 !important;
  border-color: #979797 !important;
  opacity: 1 !important;
}

.btn-danger:disabled,
.btn-danger.disabled,
.btn-danger:visited {
  background-color: #dc3545 !important;
  border-color: #979797 !important;
  opacity: 0.5 !important;
}
