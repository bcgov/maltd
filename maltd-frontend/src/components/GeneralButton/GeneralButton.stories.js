/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import GeneralButton from "./GeneralButton";

const generalButton = {
  type: "submit",
  color: "success",
  disabled: true,
  block: false,
  active: false,
  outline: false,
  placeholder: "GeneralButton"
};

storiesOf("GeneralButton", module)
  .add("default", () => <GeneralButton generalButton={generalButton} />)
  .add("valid", () => (
    <GeneralButton generalButton={{ ...generalButton, disabled: false }} />
  ))
  .add("invalid", () => (
    <GeneralButton generalButton={{ ...generalButton, color: "danger" }} />
  ));
