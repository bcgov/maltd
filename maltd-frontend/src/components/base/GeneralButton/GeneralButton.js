/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { Button } from "reactstrap";
import PropTypes from "prop-types";
import "./GeneralButton.css";

export default function GeneralButton({
  generalButton: {
    type,
    color,
    disabled,
    block,
    active,
    outline,
    label,
    styling
  }
}) {
  return (
    <Button
      className={styling}
      type={type}
      color={color}
      disabled={disabled}
      block={block}
      active={active}
      outline={outline}
    >
      {label}
    </Button>
  );
}

GeneralButton.propTypes = {
  generalButton: PropTypes.shape({
    styling: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
    block: PropTypes.bool.isRequired,
    active: PropTypes.bool.isRequired,
    outline: PropTypes.bool.isRequired,
    label: PropTypes.any.isRequired
  }).isRequired
};
