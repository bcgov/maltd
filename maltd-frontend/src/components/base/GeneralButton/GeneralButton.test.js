/* eslint-disable react/jsx-filename-extension */
import React from "react";
import renderer from "react-test-renderer";
import GeneralButton from "./GeneralButton";

describe("General Button", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <GeneralButton
        generalButton={{
          type: "submit",
          color: "primary",
          disabled: true,
          label: "Find"
        }}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
