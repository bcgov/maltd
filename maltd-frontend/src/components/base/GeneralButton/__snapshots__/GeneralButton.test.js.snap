// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Button Component renders as expected 1`] = `
<button
  aria-label={null}
  className="general-button my-2 btn btn-primary disabled"
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  Find
</button>
`;
